### DISCUSSION TITLE
How to set state as dynamic value for a class?

### DISCUSSION THREAD

**Discussion by alamenai** [tailwindlabs/tailwindcss#12675] [Help] [answered]
I have a dynamic rectangle that moves from left to right based on the received props  called `rangePoint`  from the component.

The issue is when I use this prop inside as value of `left` class, this rectangle does not change its position. 

However, if I use it inside `style` , it works well

### It does not move

```tsx
      <div className={`absolute -top-[0px] left-[${rangePoint + "%"}] w-[2px] h-8 bg-pink-500 rounded-sm`} />
```

### It moves

```tsx
  <div className={`absolute -top-[0px] w-[2px] h-8 bg-pink-500 rounded-sm`} style={{ left: rangePoint + "%" }} />
```

**Comment by wongjn** [comment]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
>
> **Don’t construct class names dynamically**
>
> ```vue
> <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> ```
> In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> Instead, make sure any class names you’re using exist in full:
>
> **Always use complete class names**
>
> ```vue
> <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> ```

**Comment by fordaaron** [comment]
You can define `rangePoint` as a css variable and then reference that value with Tailwind as an arbitrary value.
```jsx
<div className={`absolute -top-[0px] left-[var(--range-point)] w-[2px] h-8 bg-pink-500 rounded-sm`} style={{ "--range-point": rangePoint + "%" }} />
```

**Reply by wongjn** [reply, marked as answer]
You can also use the `style` attribute to set `left` without a CSS variable:

```html
<div … style={{ left: rangePoint + "%" }} />
```

**Reply by fordaaron** [reply]
Yes, that is what @MenaiAla had in his original example.

The benefit of using a css variable here would be the ability to use Tailwind pseudo-classes and media queries. Inlining it directly with `style` would mean that it could not be overridden unless `!important` was used.

**Reply by wongjn** [reply]
Yes, exactly as @MenaiAla had in their original example. As far as I can see, they are not attempting to use a variant with the `left-*` class so it seems the CSS variable is unnecessary and adds an extra rule to the CSS output.

**Comment by alamenai** [comment]
Yes, @wongjn , we used style prop in our component and it resolved the issue we added a comment beside it to explain the reason behind the decision.
